stdlib/public/core/String.swift

```diff
@@ -583,28 +583,8 @@ internal protocol _HasContiguousBytes {
   ) rethrows -> R
 }
 extension Array: _HasContiguousBytes {}
-extension UnsafeBufferPointer: _HasContiguousBytes {
-  @inlinable
-  @inline(__always)
-  func withUnsafeBytes<R>(
-    _ body: (UnsafeRawBufferPointer) throws -> R
-  ) rethrows -> R {
-    let ptr = UnsafeRawPointer(self.baseAddress._unsafelyUnwrappedUnchecked)
-    let len = self.count &* MemoryLayout<Element>.stride
-    return try body(UnsafeRawBufferPointer(start: ptr, count: len))
-  }
-}
-extension UnsafeMutableBufferPointer: _HasContiguousBytes {
-  @inlinable
-  @inline(__always)
-  func withUnsafeBytes<R>(
-    _ body: (UnsafeRawBufferPointer) throws -> R
-  ) rethrows -> R {
-    let ptr = UnsafeRawPointer(self.baseAddress._unsafelyUnwrappedUnchecked)
-    let len = self.count &* MemoryLayout<Element>.stride
-    return try body(UnsafeRawBufferPointer(start: ptr, count: len))
-  }
-}
+extension UnsafeBufferPointer: _HasContiguousBytes {}
```

Comment:
@milseman I presume the _HasContiguousBytes needs to remain for ABI stability. But the functionality was moved to `ContiguousCollection`. Do you foresee any issue with this?

Reply:
(We discussed this a bit yesterday — the primary use here for `_HasContiguousBytes` is as an existential, so, `ContiguousCollection` can't be used directly. However, it might be possible to parent `ContiguousCollection` on something like this which exposes the functionality anyway.)